kong/db/dao/init.lua

```diff
@@ -909,14 +918,14 @@ local function generate_foreign_key_methods(schema)
           return nil, tostring(err_t), err_t
         end
 
-        entity, err_t = run_hook("dao:delete_by:post",
-                                 entity,
-                                 self.schema.name,
-                                 options,
-                                 entity.ws_id)
+        entity, err_t = run_hook("dao:delete_by:post", entity, self.schema.name, options, entity.ws_id)
         if not entity then
           return nil, tostring(err_t), err_t
         end
+        err_t = run_hook_cascade_delete("dao:delete_by:post", cascade_entries, options)
```

Comment:
A bit of historical context.
The way this 'hook system' was designed to work:
1. Core has 'named hooking points' - marked by `run_hook`
2. Enterprise leverages it by registering named hooks (which run in places specified by 1.)

Is there a need for a new hooking point here, where `run_hook_cascade_delete` is called? If so, can it be made like other ones, without extra logic, and the hook registered elsewhere?

Reply:
I could introduce a `dao:delete_by:post_cascade` hook. I find it difficult to answer whether we "need" to.

We could also modify the existing `dao:delete_by:post` hook to accept the cascade_entries - this was one of my original implementations. I do not understand the benefit of adding a new hook as opposed to modifying the existing one. If the hooks are there to support EE, and I know the required EE logic is that we decrement the `workspace_entity_counters` for cascade_entries, why not modify the existing hook? Why introduce additional complexity with a completely new hook that needs to be called right after the existing hook?

I understand what we need to do - perform the same decrementing logic for the parent entity as well as cascade delete entities. It seems to me we keep circling around different implementations that, IMO, are coding style differences. I defer to developers with more tenure on style issues which is why I don't feel like I can answer whether we need to introduce a new hook.

Reply:
@tyler-ball and I will move this discussion over to Slack / Zoom